test/Microsoft.Azure.Devices.Edge.Test.Common/Platform.cs

```diff
@@ -1,27 +1,116 @@
 
-        // TODO: download installer script from aka.ms if user doesn't pass installerPath in Windows
-        public static IEdgeDaemon CreateEdgeDaemon(Option<string> installerPath) => IsWindows()
-            ? new Windows.EdgeDaemon(installerPath.Expect(() => new ArgumentException()))
-            : new Linux.EdgeDaemon() as IEdgeDaemon;
+        public static async Task<EdgeCertificates> GenerateEdgeCertificatesAsync(
+            string deviceId,
+            string scriptPath,
+            (string name, string args) command,
+            CancellationToken token)
+        {
+            await Profiler.Run(
+                async () =>
+                {
+                    string[] output = await Process.RunAsync(command.name, command.args, token);
+                    Log.Verbose(string.Join("\n", output));
+                },
+                "Created certificates for edge device");
 
-        public static string GetConfigYamlPath() => IsWindows()
-            ? Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData) + @"\iotedge\config.yaml"
-            : "/etc/iotedge/config.yaml";
+            var files = new[]
+            {
+                $"certs/iot-edge-device-{deviceId}-full-chain.cert.pem",
+                $"private/iot-edge-device-{deviceId}.key.pem",
+                "certs/azure-iot-test-only.root.ca.cert.pem"
+            };
+
+            files = NormalizeFiles(files, scriptPath).ToArray();
+
+            return new EdgeCertificates(files[0], files[1], files[2]);
+        }
+
+        public static async Task<LeafCertificates> GenerateLeafCertificatesAsync(
+            string leafDeviceId,
+            string scriptPath,
+            (string name, string args) command,
+            CancellationToken token)
+        {
+            await Profiler.Run(
+                async () =>
+                {
+                    string[] output = await Process.RunAsync(command.name, command.args, token);
+                    Log.Verbose(string.Join("\n", output));
+                },
+                "Created certificates for leaf device");
+
+            var files = new[]
+            {
+                $"certs/iot-device-{leafDeviceId}-full-chain.cert.pem",
+                $"private/iot-device-{leafDeviceId}.key.pem"
+            };
+
+            files = NormalizeFiles(files, scriptPath).ToArray();
+
+            return new LeafCertificates(files[0], files[1]);
+        }
+
+        public static async Task InstallRootCertificateAsync(
+            string scriptPath,
+            (string name, string args) command,
+            CancellationToken token)
+        {
+            string[] output = await Process.RunAsync(command.name, command.args, token);
+            Log.Verbose(string.Join("\n", output));
+
+            var files = new[]
+            {
+                "certs/azure-iot-test-only.root.ca.cert.pem",
```

Comment:
Add comments that these files are generated from which script.

Reply:
With my latest commits, the script name (certGen.sh or ca-certs.ps1) is hard-coded into the platform-specific cert functions, so it's not ambiguous anymore (i.e., the caller can't pass in some totally unrelated script). The code is not clearer about which scripts generate these files.